mesop/components/select/select.ts

```diff
@@ -12,24 +12,39 @@ import {
 } from 'mesop/mesop/components/select/select_jspb_proto_pb/mesop/components/select/select_pb';
 import {Channel} from '../../web/src/services/channel';
 import {formatStyle} from '../../web/src/utils/styles';
+import {FormsModule} from '@angular/forms';
 
 @Component({
   templateUrl: 'select.ng.html',
   standalone: true,
-  imports: [MatSelectModule],
+  imports: [MatSelectModule, FormsModule],
 })
 export class SelectComponent {
   @Input({required: true}) type!: Type;
   @Input() key!: Key;
   @Input() style!: Style;
   private _config!: SelectType;
 
+  selectedOptions: string[] = [];
+  private initialSelectOptions: string[] = [];
+
   constructor(private readonly channel: Channel) {}
 
   ngOnChanges() {
     this._config = SelectType.deserializeBinary(
       this.type.getValue() as unknown as Uint8Array,
     );
+
+    // In order to preserve backwards compatibility, only update values if the incoming
+    // values change.
+    //
+    // This is to preserve the existing behavior that allows the component to behave
+    // reasonably even if the value property is set/updated.
+    const values = this._config.getValueList();
+    if (this.initialSelectOptions !== values) {
```

Comment:
not sure if this working as intended because in JS `!==` is checking for referential equality and not value equality, i.e.

```
var a = [1]
var b = [1]
a !== b
```

note: this is different than python

Reply:
Oh dang. Can't believe I've been using javascript for this long and never knew about this behavior. Updated.